src/libraries/System.Security.Cryptography/tests/X509Certificates/ExtensionsTests/SubjectKeyIdentifierTests.cs

```diff
@@ -148,22 +153,217 @@ public static void DecodeFromBER()
+            {
+                key,
+                parameters,
+                X509SubjectKeyIdentifierHashAlgorithm.Sha384,
+                "0430150E43FE7ACE471CDB3910809145AD44B5B7E641A0364D608A1C106C9AD47963BAFE05E431B7782D791DE1B7E25F69DA",
+                "150E43FE7ACE471CDB3910809145AD44B5B7E641A0364D608A1C106C9AD47963BAFE05E431B7782D791DE1B7E25F69DA",
+            };
+
+            yield return new object[]
+            {
+                key,
+                parameters,
+                X509SubjectKeyIdentifierHashAlgorithm.Sha512,
+                "0440493CF4FC4B5CF15C17D5FCF4F85CFD1CFBCF29BEA538B8063733922A43693FEECAE70A11BEE932E23C32350C1F624DB16962A6AE6EF4B29BB3BFAD838048006F",
+                "493CF4FC4B5CF15C17D5FCF4F85CFD1CFBCF29BEA538B8063733922A43693FEECAE70A11BEE932E23C32350C1F624DB16962A6AE6EF4B29BB3BFAD838048006F",
+            };
+
+            yield return new object[]
+            {
+                key,
+                parameters,
+                X509SubjectKeyIdentifierHashAlgorithm.ShortSha256,
+                "04149F64A747E1B97F131FABB6B447296C9B6F0201E7",
+                "9F64A747E1B97F131FABB6B447296C9B6F0201E7",
+            };
+
+            yield return new object[]
+            {
+                key,
+                parameters,
+                X509SubjectKeyIdentifierHashAlgorithm.ShortSha384,
+                "04145A667D62430A8C253EBAE433333904DC6E1D41DC",
+                "5A667D62430A8C253EBAE433333904DC6E1D41DC",
+            };
+
+            yield return new object[]
+            {
+                key,
+                parameters,
+                X509SubjectKeyIdentifierHashAlgorithm.ShortSha512,
+                "0414A7C976DB1723ADB41274178DC82E9B777941AB20",
+                "A7C976DB1723ADB41274178DC82E9B777941AB20",
+            };
+        }
+
         private static void EncodeDecode(
             byte[] certBytes,
             X509SubjectKeyIdentifierHashAlgorithm algorithm,
             bool critical,
             byte[] expectedDer,
             string expectedIdentifier)
         {
-            PublicKey pk;
-
             using (var cert = new X509Certificate2(certBytes))
             {
-                pk = cert.PublicKey;
+                EncodeDecodePublicKey(cert.PublicKey, algorithm, critical, expectedDer, expectedIdentifier);
             }
+        }
 
-            X509SubjectKeyIdentifierExtension ext =
-                new X509SubjectKeyIdentifierExtension(pk, algorithm, critical);
+        private static void EncodeDecodeSubjectPublicKeyInfo(
+            byte[] spkiBytes,
+            X509SubjectKeyIdentifierHashAlgorithm algorithm,
+            bool critical,
+            byte[] expectedDer,
+            string expectedIdentifier)
+        {
+            PublicKey publicKey = PublicKey.CreateFromSubjectPublicKeyInfo(spkiBytes, out _);
+            EncodeDecodePublicKey(publicKey, algorithm, critical, expectedDer, expectedIdentifier);
+        }
+
+
```

Comment:
Nit: too many blank lines.  But since this is the only thing it's probably not worth re-spinning CI over.